yt/frontends/amrvac/datfile_utils.py

```diff
@@ -151,17 +151,12 @@ def get_single_block_data(istream, byte_offset, block_shape):
 
 def get_single_block_field_data(istream, byte_offset, block_shape, field_idx):
     """retrieve a specific block (ONE field) from a datfile"""
-    istream.seek(byte_offset)
-
     # compute byte size of a single field
     field_shape = block_shape[:-1]
     fmt = ALIGN + np.prod(field_shape) * "d"
     byte_size_field = struct.calcsize(fmt)
 
-    # Read actual data
-    istream.seek(byte_size_field * field_idx, 1)  # seek forward
-    d = struct.unpack(fmt, istream.read(struct.calcsize(fmt)))
-
-    # Fortran ordering
-    block_field_data = np.reshape(d, field_shape, order="F")
-    return block_field_data
+    istream.seek(byte_offset + byte_size_field * field_idx)
+    data = np.fromfile(istream, "=f8", count=np.prod(field_shape))
```

Comment:
Are you sure you want `=` here?  Any chance endianness will be different on generating/accepting systems?

Reply:
This module was adapted from a python script that was meant to be manually edited by users, so the endianness is actually hardcoded globally as `ALIGN = "="` (see the top lines of this module), and apparently no one ever complained about that since the frontend was released 2 yrs ago, but I agree that's there's a chance of failure here. If you have any advice on how to make this more robust I'll gladly hear them.

Reply:
Good point.. `ALIGN = "="` has been hardcoded in literally _every_ script that read in amrvac datfile data over the last few years and nobody ever complained about it, so I don't foresee any issues with this :)